File: common/src/main/java/com/datastrato/gravitino/dto/requests/GetFilesetContextRequest.java
/*
 * Copyright 2024 Datastrato Pvt Ltd.
 * This software is licensed under the Apache License version 2.
 */
package com.datastrato.gravitino.dto.requests;

import com.datastrato.gravitino.file.ClientType;
import com.datastrato.gravitino.file.FilesetDataOperation;
import com.datastrato.gravitino.file.SourceEngineType;
import com.datastrato.gravitino.rest.RESTRequest;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.google.common.base.Preconditions;
import java.util.Map;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.ToString;

/** Request to get a fileset context. */
@Getter
@EqualsAndHashCode
@NoArgsConstructor(force = true)
@AllArgsConstructor
@Builder
@ToString
public class GetFilesetContextRequest implements RESTRequest {
  @JsonProperty("subPath")
  private String subPath;

  @JsonProperty("operation")
  private FilesetDataOperation operation;

  @JsonProperty("clientType")
  private ClientType clientType;

  @JsonProperty("ip")
  private String ip;

  @JsonProperty("sourceEngineType")
  private SourceEngineType sourceEngineType;

  @JsonProperty("appId")
  private String appId;

  @JsonProperty("extraInfo")
  private Map<String, String> extraInfo;

  @Override
  public void validate() throws IllegalArgumentException {
    Preconditions.checkArgument(subPath != null, "\"subPath\" field cannot be null");
    Preconditions.checkArgument(
        operation != null, "\"operation\" field is required and cannot be null");
    Preconditions.checkArgument(
        clientType != null, "\"clientType\" field is required and cannot be null");
    Preconditions.checkArgument(
        sourceEngineType != null, "\"sourceEngineType\" field is required and cannot be null");
  }
}

File: clients/client-python/gravitino/dto/requests/get_fileset_context_request.py
"""
Copyright 2024 Datastrato Pvt Ltd.
This software is licensed under the Apache License version 2.
"""

from typing import Dict
from dataclasses import dataclass, field
from dataclasses_json import config

from gravitino.api.client_type import ClientType
from gravitino.api.fileset_data_operation import FilesetDataOperation
from gravitino.api.source_engine_type import SourceEngineType
from gravitino.rest.rest_message import RESTRequest


@dataclass
class GetFilesetContextRequest(RESTRequest):
    """Request to represent to get a fileset context."""

    _sub_path: str = field(metadata=config(field_name="subPath"))
    _operation: FilesetDataOperation = field(metadata=config(field_name="operation"))
    _client_type: ClientType = field(metadata=config(field_name="clientType"))
    _ip: str = field(metadata=config(field_name="ip"))
    _source_engine_type: SourceEngineType = field(
        metadata=config(field_name="sourceEngineType")
    )
    _app_id: str = field(metadata=config(field_name="appId"))
    _extra_info: Dict[str, str] = field(metadata=config(field_name="extraInfo"))

    def __init__(
        self,
        sub_path: str,
        operation: FilesetDataOperation,
        client_type: ClientType,
        source_engine_type: SourceEngineType,
        ip: str = None,
        app_id: str = None,
        extra_info: Dict[str, str] = None,
    ):
        self._sub_path = sub_path
        self._operation = operation
        self._client_type = client_type
        self._ip = ip
        self._source_engine_type = source_engine_type
        self._app_id = app_id
        self._extra_info = extra_info
        self.validate()

    def validate(self):
        assert self._sub_path is not None, "subPath is required"
        assert self._operation is not None, "operation is required"
        assert self._client_type is not None, "clientType is required"
        assert self._source_engine_type is not None, "sourceEngineType is required"
